### ISSUE TITLE
Repo import stuck at "Cloning...." stage

### ISSUE THREAD

**Issue by rubenvdham** [GitJournal/GitJournal#501]
## Environment

**App version:** 1.76.0 
**Device information:**
 - OS name and version: Android 11
 - Manufacturer: Oneplus
 - Model: 5

## Description
Stays Cloning forever after adding exising repo.

**Expected behavior:**
Repo cloned, I can use it from the last commit

**Current behavior:**

Stays on the "Cloning..." screen

## Steps to reproduce

1. start up app for first time
2. Add git repo manually (Gitlab)
3. Add clone URL
4. Generate public private key pair
5. Add public key to deployment keys
6. Press next
7. "Cloning...."

It does give me a message when the public key is not added to the deployment keys when I tried that for debugging, so it seems that the git repo URL is correct and working. The clone process/Activity never finishes however...

It also happens:
- When I choose automatic import using Gitlab
- When I use Github as git host, using the same repo, using automatic import

Unfortunately no logs (yet)

**Comment by rubenvdham** [comment]
UPDATE: This also happens when using "Setup Automatically", Granting Gitjournal access to Gitlab, SSH key, adding it, and again, stuck at Cloning..."

I do have enough diskspace.

**Comment by janitha** [comment]
I'm facing the same issue, both with gitea instance (1.12.6) as and also github, and I tried both generating a ssh key as well as importing an existing one. 

Version: GitJournal 1.76.0+2417
Platform: android
isPro: true

```json
{"t":1621032698611,"l":"d","msg":"githostsetup_button_click Clone Repo","stack":[{"uri":"unparsed","member":"null","isCore":false,"library":"unparsed","location":"unparsed"}]}
{"t":1621032698611,"l":"d","msg":"Event.GitHostSetupButtonClick","stack":[{"uri":"unparsed","member":"null","isCore":false,"library":"unparsed","location":"unparsed"}],"p":{"text":"Clone Repo","icon_url":""}}
{"t":1621032698611,"l":"i","msg":"RepoPath: /data/user/0/io.gitjournal.gitjournal/app_flutter/journal","stack":[{"uri":"unparsed","member":"null","isCore":false,"library":"unparsed","location":"unparsed"}]}
{"t":1621032699088,"l":"w","msg":"Could not fetch git main branch","ex":"type 'Future<dynamic>' is not a subtype of type 'FutureOr<String?>' in type cast","stack":[{"uri":"unparsed","member":"null","isCore":false,"library":"unparsed","location":"unparsed"}]}
```

The error `Could not fetch git main branch` seemed interesting, so I tried changing the default branch on the remote server's repo to `main` but I was still not able to clone. 

Also, I was tailing the logs of gitea (as I pressed clone in GitJournal), I can see the ssh authentication authenticating just fine (and when I didn't have the deploykeys setup, GitJournal did immediately error out), so at least it doesn't seem like a problem with the ssh keys and authentication.

```
Accepted publickey for git from 10.42.1.1 port 35656 ssh2: RSA SHA256:fHEu/NwFTT2sscavm07NHeIlvuQzHGGoLINhyaUubmE
[Macaron] 2021-05-14 22:51:39: Started GET /api/internal/serv/command/13/janitha/test-repo-1?mode=1&verb=git-upload-pack for [::1]
[Macaron] 2021-05-14 22:51:39: Completed GET /api/internal/serv/command/13/janitha/test-repo-1?mode=1&verb=git-upload-pack 200 OK in 8.301838ms
[Macaron] 2021-05-14 22:51:39: Started POST /api/internal/ssh/13/update/24 for [::1]
[Macaron] 2021-05-14 22:51:39: Completed POST /api/internal/ssh/13/update/24 200 OK in 74.089274ms
Received disconnect from 10.42.1.1 port 35656:11: closing transport
Disconnected from user git 10.42.1.1 port 35656
```

**Comment by vHanda** [comment]
I managed to reproduce it. I see the problem. I'll fix it tomorrow and push and update. Thanks for the helping in diagnosing the issue

**Comment by vHanda** [comment]
I couldn't sleep. I've fixed the bug and released a new version `1.76.1`.

Please let me know if this still occurs.

**Comment by rubenvdham** [comment]
I was about to close/update this, as I figured it suddenly worked after last night's reboot. That was fast! I didn't see this coming, great work! 🎉🎉👌🏻
